docs/how_to_guides/miscellaneous/how_to_add_and_test_a_new_sqlalchemydataset_class.rst

```diff
@@ -0,0 +1,215 @@
+.. _how_to_guides__miscellaneous__how_to_add_and_test_a_new_sqlalchemydataset_class:
+
+How to add support for a new SQLAlchemy dialect
+===============================================
+
+This guide will help you extend the Great Expectations execution layer to work on a new SQL dialect.
+
+.. warning::
+
+   This guide is a working checklist. We improve it each time we add support for a new SQL dialect, but it still has rough edges. And adding support for a new database will probably never be a cookie-cutter operation.
+
+   If you're interested in extending Great Expectations in this way, please reach out on `Slack <greatexpectations.io/slack>`__ in the ``#contributors-contributing`` channel, and we'll work with you to get there.
+
+Steps
+-----
+
+1. Install and document dependencies
+####################################
+
+    * **Install the python dependencies for your dialect.**  While you're at it, please add a line(s) to ``requirements-dev.txt``. Examples:
+    
+        .. code-block:: bash
+        
+            pyhive[presto]>=0.6.2
+            PyAthena[SQLAlchemy]>=1.1
+            pybigquery>=0.4.15
+
+    * **Install any necessary drivers**. sqlalchemy requires special drivers for some databases. If yours is one of them, please carefully document the steps in {{document TBD}}
+    * **If possible, add docker containers for test resources.** (You can find examples of similar containers in `assets/docker/`.) Otherwise, please document the steps to set up a suitable test environment.
+
+The core team will not be able to merge your contribution until they're able to replicate your setup.
+
+
+2. Make changes to sqlalchemy_dataset.py
+########################################
+
+    * **At the top the file, add a try-except block**
+
+        .. code-block:: python
+
+            try:
+                import pyhive.sqlalchemy_presto as presto
+            except ImportError:
+                presto = None
+
+
+    * **In SqlAlchemyDataset.__init__, add an elif clause**
+
+        .. code-block:: python
+
+            elif self.engine.dialect.name.lower() == "presto":
+                self.dialect = import_module("pyhive.sqlalchemy_presto")
+
+    * **In SqlAlchemyDataset.create_temporary_table, add an elif clause**
+
+        .. code-block:: python
+
+            elif (self.engine.dialect.name.lower() == "awsathena"
+              or self.engine.dialect.name.lower() == "presto"):
+                stmt = "CREATE OR REPLACE TABLE {table_name} AS {custom_sql}".format(
+                    table_name=table_name, custom_sql=custom_sql)
+
```

Comment:
Add implementations for your dialect for all relevant _get_dialect_XXX methods

Reply:
I'm not sure what you're saying here.

Reply:
The contributor must review and usually add implementations for any method that starts with `_get_dialect` for their dialect.